tests/device_tests/test_msg_applysettings.py

```diff
@@ -74,6 +74,32 @@ def test_apply_settings_passphrase(self, client):
 
         assert client.features.passphrase_protection is True
 
+    @pytest.mark.setup_client(passphrase=False)
+    @pytest.mark.skip_t1
+    def test_apply_settings_passphrase_on_device(self, client):
+        # enable passphrase
+        with client:
+            client.set_expected_responses(EXPECTED_RESPONSES_NOPIN)
+            device.apply_settings(client, use_passphrase=True)
+
+        assert client.features.passphrase_protection is True
+
+        # enable force on device
+        with client:
+            client.set_expected_responses(EXPECTED_RESPONSES_NOPIN)
+            device.apply_settings(client, passphrase_always_on_device=True)
+
+        assert client.features.passphrase_protection is True
+        assert client.features.passphrase_always_on_device is True
+
+        # turning off the passphrase should also clear the always_on_device setting
+        with client:
+            client.set_expected_responses(EXPECTED_RESPONSES_NOPIN)
+            device.apply_settings(client, use_passphrase=False)
+
+        assert client.features.passphrase_protection is False
+        assert client.features.passphrase_always_on_device is False
```

Comment:
please add "re-enable passphrase" and check that `passphrase_always_on_device` is False afterwards as well

Reply:
Fixed